ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala

```diff
@@ -41,10 +42,21 @@ private[ftp] trait SftpOperations { _: FtpLike[SSHClient, SftpSettings] =>
 
     ssh.connect(host.getHostAddress, port)
 
-    if (credentials.password != "" && sftpIdentity.isEmpty)
-      ssh.authPassword(credentials.username, credentials.password)
+    if (credentials.password != "" && sftpIdentity.isDefined) {
+      val passwordAuth: Seq[AuthPassword] = credentials.password.map { pass =>
```

Comment:
This is weird, it will create an `AuthPassword` for every character in `credentials.password`.

Reply:
Using map to convert it to seq

Reply:
While it 'makes the types fit', I'm pretty sure it's wrong.

Could you add a testcase showing it's actually right?

Reply:
@raboof I will try to add unit tests and test cases
 

Can you elaborate more on why do you think it is wrong?

Reply:
As enno mentioned this looks like it'd create an `AuthPassword` for every character in the password, which I can't imagine is the right behavior.

Reply:
okay. I changed that. it was creating the AuthPassword multiple times which I changed.

Sorry for that oversight on myside